src/legacy/core_plugins/interpreter/server/lib/route_expression/browser.js

```diff
@@ -22,12 +22,21 @@ import uuid from 'uuid/v4';
 export const browser = ({ socket, serialize, deserialize }) => {
   // Note that we need to be careful about how many times routeExpressionProvider is called, because of the socket.once below.
   // It's too bad we can't get a list of browser plugins on the server
-  const getClientFunctions = new Promise(resolve => {
+  let getClientFunctions = new Promise(resolve => {
     socket.emit('getFunctionList');
     socket.once('functionList', resolve);
   });
 
-  return getClientFunctions.then(functions => {
+  socket.on('updateFunctionList', () => {
+    getClientFunctions = new Promise(resolve => {
+      socket.emit('getFunctionList');
```

Comment:
Can you de-dupe this into a function, instead of pasting the same code here. That way the two won't diverge. Something like this would do:

```
const getFunctionList = () => new Promise(resolve => {
    socket.emit('getFunctionList');
    socket.once('functionList', resolve);
  });

let getClientFunctions = getFunctionList();

socket.on('updateFunctionList', () => {
  getClientFunctions = getFunctionList();
});
```